apps/artemis-web/middleware/captcha.ts

```diff
@@ -0,0 +1,24 @@
+import memory from "../utils/captchaMemoryStore";
+
+const lambdaCaptcha = require('lambda-captcha')
+const SECRET = process.env.CAPTCHA_SECRET
+
+const captchaChallengeStore = {};
+
+export default function captcha(page: string) {
+    return async (req, res, next): Promise<any> => {
+        const captchaResult = lambdaCaptcha.verify(req.body.encryptedExpr, req.body.captcha, SECRET);
+        const hits = memory.getHits(req.ip);
+        if (!captchaChallengeStore[req.ip])
+            captchaChallengeStore[req.ip] = [];
+        const skipCaptcha = (page === 'login' && hits < (parseInt(process.env.CAPTCHA_TRIES ?? '4'))) || (process.env.TESTING === 'true');
```

Comment:
what if hits is undefined? Can it be or is it initialized to 0 by default?

Reply:
I think you are right. Added a fallback